### DISCUSSION TITLE
Artisan Facade is not mockable

### DISCUSSION THREAD

**Discussion by NiroDeveloper** [orchestral/testbench#412] [Q&A] [answered]
- Testbench Version: 9.2.0
- Laravel Version: 11.6.0
- PHP Version: 8.3

### Description:
It is not possible to mock the Artisan Facade to test command calls.

> Mockery\Exception: The class \Orchestra\Testbench\Console\Kernel is marked final and its methods cannot be replaced. Classes marked final can be passed in to \Mockery::mock() as instantiated objects to create a partial mock, but only if the mock is not subject to type hinting checks.

Is there a specific reason that this class is final, otherwise we could just remove the final keyword.

### Steps To Reproduce:
1. Write a unit test
2. Add `Artisan::shouldReceive("call")->with("backup:run")->andReturn(Command::SUCCESS)->once();`

**Comment by crynobone** [comment, marked as answer]
No plan for this, you can override the Console Kernel implementations based on https://packages.tools/testbench/the-basic.html#overriding-console-kernel

Alternatively, You can also create `workbench/app/Console/Kernel.php` and use `WithWorkbench` trait to automatically load your custom console kernel: https://packages.tools/testbench.html#autoloading-using-testbench-yaml
